src/main/java/com/hms_networks/americas/sc/extensions/mqtt/ConstrainedMqttManager.java

```diff
@@ -0,0 +1,929 @@
+   * @param mqttQos the QoS (quality of service) level of the MQTT client connection
+   * @param wanIpTimeout the timeout (in the specified {@code wanIpTimeoutUnit}s) to wait for a WAN
+   *     IP address to be available.
+   * @param wanIpTimeoutUnit the {@link SCTimeUnit} of the specified {@code wanIpTimeout}.
+   * @throws IllegalStateException if the specified {@code wanIpTimeout} (in the specified {@code
+   *     wanIpTimeoutUnit}s) is reached before a WAN IP address is available.
+   * @throws Exception if unable to start the {@link MqttManager} instance using the provided
+   *     parameters.
+   * @since 1.0.0
+   */
+  public ConstrainedMqttManager(
+      String mqttId,
+      String mqttHost,
+      boolean utf8Enable,
+      String mqttPort,
+      String mqttCaFilePath,
+      String mqttTlsVersion,
+      String mqttUsername,
+      String mqttPassword,
+      int mqttQos,
+      int wanIpTimeout,
+      SCTimeUnit wanIpTimeoutUnit)
+      throws Exception {
+    this.mqttId = mqttId;
+    this.mqttHost = mqttHost;
+    this.utf8Enable = utf8Enable;
+    this.mqttPort = mqttPort;
+    this.mqttCaFilePath = mqttCaFilePath;
+    this.mqttTlsVersion = mqttTlsVersion;
+    this.mqttUsername = mqttUsername;
+    this.mqttPassword = mqttPassword;
+    this.mqttQos = mqttQos;
+    this.mqttSubscriptions = new ArrayList();
+    this.mqttSubscribed = false;
+
+    // Wait for WAN IP (if enabled)
+    if (wanIpTimeout == WAIT_FOR_WAN_IP_TIMEOUT_INDEFINITE) {
+      SCAppManagement.waitForWanIp();
+    } else if (wanIpTimeout != WAIT_FOR_WAN_IP_TIMEOUT_DISABLED && wanIpTimeoutUnit != null) {
+      SCAppManagement.waitForWanIp(wanIpTimeout, wanIpTimeoutUnit);
+    }
+
+    // Check for WAN IP
+    if (!SCAppManagement.deviceHasWanIP()) {
+      throw new IllegalStateException("Device does not have WAN IP");
+    }
+  }
+
+  /**
+   * Builds a new, wrapped {@link MqttManager} instance using the current MQTT parameters.
+   *
+   * @return the new, wrapped {@link MqttManager} instance
+   * @throws Exception if unable to create a new, wrapped {@link MqttManager} instance
+   * @since 1.0.0
+   */
+  private MqttManager buildMqttManager() throws Exception {
+    // Return new MQTT manager instance which uses the internal methods
+    return new MqttManager(mqttId, mqttHost, utf8Enable) {
+      /**
+       * Manages errors (exceptions) that occur in the MQTT manager.
+       *
+       * @param throwable exception to be handled
+       * @since 1.0.0
+       */
+      public void onError(Throwable throwable) {
+        internalError(throwable);
+      }
+
+      /**
+       * Manages received MQTT messages.
+       *
+       * @param mqttMessage received MQTT message
+       * @since 1.0.0
+       */
+      public void onMessage(MqttMessage mqttMessage) {
+        internalMessage(mqttMessage);
+      }
+
+      /**
+       * Method for processing or performing tasks on the looping MQTT thread.
```

Comment:
This is a bit confusing. Is it processing, performing, or both. 

Could this be simplified to "Method for handling tasks in the looping MQTT thread"?

Reply:
Updated